pyaerocom/aeroval/experiment_output.py

```diff
@@ -220,7 +222,8 @@ def _sync_heatmaps_with_menu_and_regions(self) -> None:
                             hm_data = self._check_hm_all_regions_avail(all_regions, hm_data)
                             hm[vardisp][obs][vert_code][mod][modvar] = hm_data
 
-            self.avdb.put_by_uuid(hm, fp)
+            with self.avdb.lock():
+                self.avdb.put_by_uri(hm, fp)
```

Comment:
`with self.avdb.lock():` must start before line 205 `data = self.avdb.get_by_uri(fp)` otherwise data might be updated by another process.